### PR TITLE
A couple minor improvements for tidy error handling.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/tools/tidy/src/bins.rs [hunk 1]
```diff
@@ -62,7 +62,7 @@ pub fn check(path: &Path, bad: &mut bool) {
                 });
             let path_bytes = rel_path.as_os_str().as_bytes();
             if output.status.success() && output.stdout.starts_with(path_bytes) {
-                println!("binary checked into source: {}", file.display());
+                tidy_error!("binary checked into source: {}", file.display());
                 *bad = true;
```
Comment (petrochenkov) [inline src/tools/tidy/src/bins.rs]: `*bad = true` should probably a part of `tidy_error`

**Comment by alexcrichton** [comment]
r=me with @petrochenkov's comment

**Comment by frewsxcv** [comment]
Going to wait for https://github.com/rust-lang/rust/pull/40694 to merge.

**Comment by frewsxcv** [comment]
Comments have been addressed.

**Comment by frewsxcv** [comment]
https://github.com/rust-lang/rust/issues/40546

@bors retry

**Pull request merged**